test/evolution_test.rb

```diff
@@ -46,8 +46,9 @@ class EvolutionTest < ActiveSupport::TestCase
 
     assert_equal [record], child.parents
     assert_equal [child], record.children
-    assert_equal [record, child].sort, child.ancestors.sort
-    assert_equal [record, child].sort, record.descendants.sort
+
+    assert_equal [record], child.ancestors
```

Comment:
Should `ancestors` and `descendants` include `self`? It seems they should not but the previous test suite (with acts_as_dag 1.x) was passing.

https://github.com/rrn/acts_as_dag/blob/637a9c31dbfbb33ddff30cbd73d3d3db7c6c44af/spec/acts_as_dag_spec.rb#L145-L147
https://github.com/rrn/acts_as_dag/blob/637a9c31dbfbb33ddff30cbd73d3d3db7c6c44af/spec/acts_as_dag_spec.rb#L72-L74

I've tried to do a bit of archeology on this but haven't found the exact moment this regression happened.  I thought I'd check in with you first.

It looks like this spec [changed in acts_as_dag in this commit](https://github.com/rrn/acts_as_dag/commit/0e30de988f6fa85ef26cf301f2da09a570ab07b6#diff-549165fe1c72cf1f1e1e527d06e10267a05c71a4c144796752bfd89eee9163a0)

I'm happy to dig further to try to find the exact commit where things changed. Or is it ok to just go ahead with `ancestors` and `descendants` not including `self`?

Reply:
I think it's fine to not. On the face of it, it seems wrong for a method called `descendants` to include self.

cc @njakobsen